### PR TITLE
Issue/38837

### LINKED ISSUE #38837: onyx_bgp - Support purge option to not delete existing bgp neighbors
##### ISSUE TYPE

 - Feature Idea
 

##### COMPONENT NAME

onyx_bgp 
Adding neighbor(s) is a useful common task, adding a purge flag in the module and only if true will remove other neighbors, 
##### ANSIBLE VERSION

```
ansible 2.6.0 (devel db80504839) last updated 2018/04/12 16:53:04 (GMT +300)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/ansible/ansible/lib/ansible
  executable location = /root/ansible/ansible/bin/ansible
  python version = 2.7.5 (default, Aug  2 2016, 04:20:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]
```

##### CONFIGURATION

HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False
PERSISTENT_CONNECT_TIMEOUT(/etc/ansible/ansible.cfg) = 300

##### OS / ENVIRONMENT

N/A
##### STEPS TO REPRODUCE

add the purge flag for the onyx_bgp module

```yaml
---
- name: bgp neighbors
  onyx_bgp:
     as_number: 65000
     router_id: 192.168.168.168
     neighbors:
        - remote_as: 65002
          neighbor: 192.168.255.3
     purge: no
     state: present
     networks:
       - 192.168.168.168/32
       -  192.168.100.0/24
```

##### EXPECTED RESULTS

when purge is set to no , the task won't delete the existing bgp neighbors on the Onyx device
##### ACTUAL RESULTS

```

```

### PR DESCRIPTION
**Pull request by anasbadaha**
Signed-off-by: Anas Badaha <anasb@mellanox.com>

##### SUMMARY
Fixes #38837

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/network/onyx/onyx_bgp.py
test/units/modules/network/onyx/test_onyx_bgp.py

##### ADDITIONAL INFORMATION

```
ansible 2.8.0.dev0 (issue/38837 38deff2294) last updated 2018/12/13 09:53:23 (GMT +000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /.autodirect/mtrswgwork/anasb/ansible_dev8/lib/ansible
  executable location = /.autodirect/mtrswgwork/anasb/ansible_dev8/bin/ansible
  python version = 2.7.5 (default, Jul 13 2018, 13:06:57) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

```

### CODE REVIEW

**Comment by anasbadaha** [comment]
Hi @samerd,

this is the fix for the bgp bug please let me know if it's ok for you ? also please take a look on failed test in shippable.

File: lib/ansible/modules/network/onyx/onyx_bgp.py [hunk 1]
```diff
@@ -49,9 +49,9 @@
...
     default: present
     choices: ['present', 'absent']
-  neighbors_purge:
+  purge:
     description:
-      - will Purge all neighbors when it is True.
+      - will remove all neighbors when it is True.
     type: bool
     default: false
 """
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_bgp.py:58]: version_added: 2.8

Comment (samerd) [reply lib/ansible/modules/network/onyx/onyx_bgp.py:58]: needs to be added in the purge attributes

**Comment by anasbadaha** [comment]
Hi @ganeshrn this PR is approved by @samerd please merge it

**Review by ganeshrn — APPROVED** [review]

**Comment by ganeshrn** [comment]
@anasbadaha Thank you!

**Pull request merged**